data_steward/gcloud/gcs/__init__.py

```diff
@@ -18,6 +20,40 @@ class StorageClient(Client):
     See https://googleapis.dev/python/storage/latest/client.html
     """
 
+    def get_bucket_items_metadata(self, bucket: Bucket) -> list:
+        """
+        Given a bucket, iterate through it's contents and pull out each objects
+        metadata
+        :param bucket: Bucket to iterate through
+        :return: a list of dicts containing metadata
+        """
+
+        blobs: list = list(self.list_blobs(bucket))
+        metadata: list = [self.get_blob_metadata(blob) for blob in blobs]
+        return metadata
+
+    def get_blob_metadata(self, blob: Blob) -> dict:
+        """
+        Gather and ship a blob's metadata in dictionary form.
+        Note: Date and times are in the python stl format.  They are datetime objects.
+        """
+        metadata: dict = {
```

Comment:
Do you think this should have a call to the client if it received a blob object that is not instantiated with the HTTP request?
```suggestion
        if blob.id == None:
            blob = self.client.get_bucket(blob.bucket.name).get_blob(blob.name)
        metadata: dict = {
```

Reply:
Will implement